### PR TITLE
Refurbish argparse options

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Remove unused, add as method/class attributes the ones on use.

### CODE REVIEW

File: pyfstat/helper_functions.py [hunk 1]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
...
-        action="store_true",
-        help="Only generate the setup file, don't run",
-    )
-    fu_parser.add_argument(
-        "--no-template-counting",
-        action="store_true",
-        help="No counting of templates, useful if the setup is predefined",
-    )
-    parser.add_argument(
-        "-N",
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:84]: this seems to actually be used in `FrequencyModulatedArtifactWriter`

File: pyfstat/helper_functions.py [hunk 2]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
         action="store_true",
         help="Decrease output verbosity [logging.WARNING]",
     )
-    parser.add_argument(
-        "--no-interactive", help="Don't use interactive", action="store_true"
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:61]: Seems to have been unused already, so :+1:

File: pyfstat/helper_functions.py [hunk 3]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
...
         "--clean",
         action="store_true",
         help="Force clean data, never use cached data",
     )
-    fu_parser = parser.add_argument_group(
-        "follow-up options", "Options related to MCMCFollowUpSearch"
-    )
-    fu_parser.add_argument(
-        "-s",
-        "--setup-only",
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:74]: replaced by class option :+1:

File: pyfstat/helper_functions.py [hunk 4]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
...
-    parser.add_argument(
-        "--no-interactive", help="Don't use interactive", action="store_true"
-    )
     parser.add_argument(
         "-c",
         "--clean",
         action="store_true",
         help="Force clean data, never use cached data",
     )
-    fu_parser = parser.add_argument_group(
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:69]: bo idea if this ever did anything :shrug:

File: pyfstat/helper_functions.py [hunk 5]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
...
-        "follow-up options", "Options related to MCMCFollowUpSearch"
-    )
-    fu_parser.add_argument(
-        "-s",
-        "--setup-only",
-        action="store_true",
-        help="Only generate the setup file, don't run",
-    )
-    fu_parser.add_argument(
-        "--no-template-counting",
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:79]: replaced by class option :+1:

File: pyfstat/helper_functions.py [hunk 6]
```diff
@@ -57,39 +56,13 @@ def set_up_command_line_arguments():
...
-    parser.add_argument(
-        "-N",
-        type=int,
-        default=3,
-        metavar="N",
-        help="Number of threads to use when running in parallel",
-    )
-    parser.add_argument("unittest_args", nargs="*")
     args, unknown = parser.parse_known_args()
-    sys.argv[1:] = args.unittest_args
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:92]: don't think we ever used this for anything :shrug: Ok if tests pass.

File: pyfstat/helper_functions.py (hunk 1 above)
Comment (Rodrigo-Tenorio) [reply pyfstat/helper_functions.py:84]: Ups, forgot to check that one :facepalm: thanks

File: pyfstat/helper_functions.py (hunk 4 above)
Comment (Rodrigo-Tenorio) [reply pyfstat/helper_functions.py:69]: It did. `no-template-counting` was my personal hell for a while :wink:

File: pyfstat/helper_functions.py [hunk 7]
```diff
@@ -57,39 +56,7 @@ def set_up_command_line_arguments():
         action="store_true",
         help="Decrease output verbosity [logging.WARNING]",
     )
-    parser.add_argument(
-        "--no-interactive", help="Don't use interactive", action="store_true"
-    )
-    parser.add_argument(
-        "-c",
-        "--clean",
```
Comment (dbkeitel) [inline pyfstat/helper_functions.py:65]: looks sufficiently replaced by class options :+1:

Comment (dbkeitel) [reply pyfstat/helper_functions.py:65]: ...or not: `AttributeError: 'GridGlitchSearch' object has no attribute 'clean'`

Comment (Rodrigo-Tenorio) [reply pyfstat/helper_functions.py:65]: Should be fixed by now.

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**